src/super_gradients/training/transforms/transforms.py

```diff
@@ -756,88 +761,59 @@ class DetectionTargetsFormatTransform(DetectionTransform):
-
-        _, h, w = image.shape
-
-        def _format_target(targets_in):
-            if label_first_in_input:
-                labels, boxes = targets_in[:, 0], targets_in[:, 1:]
-            else:
-                boxes, labels = targets_in[:, :4], targets_in[:, 4]
-
-            if convert2cxcy:
-                boxes = xyxy2cxcywh(boxes)
-            elif convert2xyxy:
-                boxes = cxcywh2xyxy(boxes)
-
-            if normalize:
-                boxes[:, 0] = boxes[:, 0] / w
-                boxes[:, 1] = boxes[:, 1] / h
-                boxes[:, 2] = boxes[:, 2] / w
-                boxes[:, 3] = boxes[:, 3] / h
-
-            elif denormalize:
-                boxes[:, 0] = boxes[:, 0] * w
-                boxes[:, 1] = boxes[:, 1] * h
-                boxes[:, 2] = boxes[:, 2] * w
-                boxes[:, 3] = boxes[:, 3] * h
-
-            min_bbox_edge_size = self.min_bbox_edge_size / max(w, h) if normalized_output else self.min_bbox_edge_size
+    def __call__(self, sample: dict) -> dict:
+        sample["target"] = self.apply_on_targets(sample["target"])
+        if "crowd_target" in sample.keys():
+            sample["crowd_target"] = self.apply_on_targets(sample["crowd_target"])
+        return sample
 
-            cxcywh_boxes = boxes if not output_xyxy_format else xyxy2cxcywh(boxes.copy())
+    def apply_on_targets(self, targets: np.ndarray) -> np.ndarray:
+        """Convert targets in input_format to output_format, filter small bboxes and pad targets"""
+        targets = self.targets_format_converter(targets)
+        targets = self.filter_small_bboxes(targets)
+        targets = self.pad_targets(targets)
+        return targets
 
-            mask_b = np.minimum(cxcywh_boxes[:, 2], cxcywh_boxes[:, 3]) > min_bbox_edge_size
-            boxes_t = boxes[mask_b]
-            labels_t = labels[mask_b]
+    def filter_small_bboxes(self, targets: np.ndarray) -> np.ndarray:
+        """Filter bboxes smaller than specified threshold."""
 
-            labels_t = np.expand_dims(labels_t, 1)
-            targets_t = np.hstack((labels_t, boxes_t)) if label_first_in_output else np.hstack((boxes_t, labels_t))
-            padded_targets = np.zeros((self.max_targets, 5))
-            padded_targets[range(len(targets_t))[: self.max_targets]] = targets_t[: self.max_targets]
-            padded_targets = np.ascontiguousarray(padded_targets, dtype=np.float32)
+        def _is_big_enough(bboxes: np.ndarray) -> np.ndarray:
+            return np.minimum(bboxes[:, 2], bboxes[:, 3]) > self.min_bbox_edge_size
 
-            return padded_targets
+        targets = filter_on_bboxes(fn=_is_big_enough, tensor=targets, tensor_format=self.output_format)
+        return targets
 
-        sample["target"] = _format_target(targets)
-        if crowd_targets is not None:
-            sample["crowd_target"] = _format_target(crowd_targets)
-        return sample
+    def pad_targets(self, targets: np.ndarray) -> np.ndarray:
```

Comment:
This padding of targets to `max_targets` should be a part of collate function not a target transform, IMO.
Since this padding is specific to YoloX. 
Yes, this is not directly related to this PR and should not prevent from merging it, but worth keeping in mind. CC @ofrimasad

Reply:
IMO `filter_small_bboxes` should also be moved to another transform (who could expect DetectionTargetsFormatTransform to also filter boxes?).
I didnt do it here to not overload more the PR, but I created a ticket for that